### PR TITLE
Mesh sync

### PR DESCRIPTION
**Pull request by KKopyscinski**

### CODE REVIEW

**Comment by KKopyscinski** [comment]
Test results from PTS v8.1.2

<details>
<summary>Test results</summary>
<pre>
MESH    MESH/NODE/PBADV/BV-01-C         PASS
MESH    MESH/NODE/PBADV/BV-02-C         PASS
MESH    MESH/NODE/PBADV/BV-03-C         PASS
MESH    MESH/NODE/PBADV/BV-04-C         PASS
MESH    MESH/NODE/PBADV/BI-01-C         PASS
MESH    MESH/NODE/PBADV/BI-02-C         PASS
MESH    MESH/NODE/PBADV/BI-03-C         PASS
MESH    MESH/NODE/PBADV/BI-04-C         PASS
MESH    MESH/NODE/MPS/BV-01-C           PASS
MESH    MESH/NODE/MPS/BV-02-C           PASS
MESH    MESH/NODE/MPS/BV-03-C           PASS
MESH    MESH/NODE/MPS/BV-04-C           PASS
MESH    MESH/NODE/MPS/BV-05-C           PASS
MESH    MESH/NODE/MPS/BV-06-C           PASS
MESH    MESH/NODE/MPS/BV-07-C           PASS
MESH    MESH/NODE/MPS/BV-08-C           PASS                  
MESH    MESH/NODE/MPS/BV-09-C           PASS                  
MESH    MESH/NODE/MPS/BV-10-C           PASS 
MESH    MESH/NODE/MPS/BI-01-C           PASS
MESH    MESH/NODE/PROV/BV-01-C          PASS
MESH    MESH/NODE/PROV/BV-07-C          PASS
MESH    MESH/NODE/PROV/BV-09-C          PASS
MESH    MESH/NODE/PROV/BV-10-C          PASS
MESH    MESH/NODE/PROV/BV-11-C          PASS
MESH    MESH/NODE/PROV/BV-12-C          PASS
MESH    MESH/NODE/PROV/BI-01-C          PASS
MESH    MESH/NODE/PROV/BI-02-C          PASS
MESH    MESH/NODE/PROV/BI-03-C          PASS
MESH    MESH/NODE/PROV/BI-13-C          PASS
MESH    MESH/NODE/PROV/BI-15-C          PASS
MESH    MESH/NODE/PROV/BI-17-C          PASS
MESH    MESH/NODE/BCN/SNB/BV-01-C       PASS
MESH    MESH/NODE/NET/BV-01-C           PASS
MESH    MESH/NODE/NET/BV-02-C           PASS
MESH    MESH/NODE/NET/BV-03-C           PASS
MESH    MESH/NODE/NET/BV-04-C           PASS
MESH    MESH/NODE/NET/BV-05-C           PASS
MESH    MESH/NODE/NET/BV-06-C           PASS
MESH    MESH/NODE/NET/BV-07-C           BTP TIMEOUT
MESH    MESH/NODE/NET/BV-08-C           PASS
MESH    MESH/NODE/NET/BV-09-C           PASS
MESH    MESH/NODE/NET/BV-10-C           PASS
MESH    MESH/NODE/NET/BV-11-C           PASS
MESH    MESH/NODE/NET/BV-12-C           PASS
MESH    MESH/NODE/NET/BV-13-C           PASS
MESH    MESH/NODE/NET/BV-14-C           INCONC
MESH    MESH/NODE/NET/BI-01-C           PASS
MESH    MESH/NODE/NET/BI-02-C           PASS
MESH    MESH/NODE/NET/BI-03-C           PASS
MESH    MESH/NODE/RLY/BV-01-C           PASS
MESH    MESH/NODE/RLY/BV-02-C           BTP TIMEOUT
MESH    MESH/NODE/RLY/BI-01-C           PASS
MESH    MESH/NODE/RLY/BI-02-C           PASS
MESH    MESH/NODE/TNPT/BV-01-C          PASS
MESH    MESH/NODE/TNPT/BV-02-C          PASS
MESH    MESH/NODE/TNPT/BV-03-C          PASS
MESH    MESH/NODE/TNPT/BV-04-C          PASS
MESH    MESH/NODE/TNPT/BV-05-C          PASS
MESH    MESH/NODE/TNPT/BV-06-C          PASS
MESH    MESH/NODE/TNPT/BV-07-C          PASS
MESH    MESH/NODE/TNPT/BV-08-C          PASS
MESH    MESH/NODE/TNPT/BV-09-C          PASS
MESH    MESH/NODE/TNPT/BV-10-C          PASS
MESH    MESH/NODE/TNPT/BV-11-C          PASS
MESH    MESH/NODE/TNPT/BV-12-C          PASS
MESH    MESH/NODE/TNPT/BV-13-C          FAIL
MESH    MESH/NODE/TNPT/BI-01-C          PASS
MESH    MESH/NODE/TNPT/BI-02-C          PASS
MESH    MESH/NODE/IVU/BV-01-C           PASS
MESH    MESH/NODE/IVU/BV-02-C           PASS
MESH    MESH/NODE/IVU/BV-03-C           PASS
MESH    MESH/NODE/IVU/BV-04-C           PTS TIMEOUT
MESH    MESH/NODE/IVU/BV-05-C           PASS
MESH    MESH/NODE/IVU/BI-01-C           PASS
MESH    MESH/NODE/IVU/BI-02-C           PASS
MESH    MESH/NODE/IVU/BI-04-C           PASS
MESH    MESH/NODE/IVU/BI-05-C           PASS
MESH    MESH/NODE/IVU/BI-06-C           PASS
MESH    MESH/NODE/KR/BV-01-C            PASS
MESH    MESH/NODE/KR/BV-02-C            PASS
MESH    MESH/NODE/KR/BV-03-C            PASS
MESH    MESH/NODE/KR/BI-01-C            PASS
MESH    MESH/NODE/KR/BI-02-C            PASS
MESH    MESH/NODE/KR/BI-03-C            PASS
MESH    MESH/NODE/KR/BI-04-C            PASS
MESH    MESH/NODE/KR/BI-05-C            PASS
MESH    MESH/NODE/FRND/FN/BV-01-C       PASS
MESH    MESH/NODE/FRND/FN/BV-02-C       PASS
MESH    MESH/NODE/FRND/FN/BV-03-C       PASS
MESH    MESH/NODE/FRND/FN/BV-04-C       PASS
MESH    MESH/NODE/FRND/FN/BV-05-C       PASS
MESH    MESH/NODE/FRND/FN/BV-06-C       PASS
MESH    MESH/NODE/FRND/FN/BV-07-C       PASS
MESH    MESH/NODE/FRND/FN/BV-08-C       PASS
MESH    MESH/NODE/FRND/FN/BV-09-C       PASS
MESH    MESH/NODE/FRND/FN/BV-10-C       PASS
MESH    MESH/NODE/FRND/FN/BV-11-C       PASS
MESH    MESH/NODE/FRND/FN/BV-12-C       INCONC
MESH    MESH/NODE/FRND/FN/BV-13-C       PASS
MESH    MESH/NODE/FRND/FN/BV-14-C       PASS
MESH    MESH/NODE/FRND/FN/BV-15-C       PASS
MESH    MESH/NODE/FRND/FN/BV-16-C       PASS
MESH    MESH/NODE/FRND/FN/BV-17-C       PASS
MESH    MESH/NODE/FRND/FN/BV-18-C       PASS
MESH    MESH/NODE/FRND/FN/BV-19-C       PASS
MESH    MESH/NODE/FRND/FN/BV-20-C       INCONC
MESH    MESH/NODE/FRND/FN/BV-21-C       PASS
MESH    MESH/NODE/FRND/FN/BV-22-C       PASS
MESH    MESH/NODE/FRND/FN/BV-23-C       PASS
MESH    MESH/NODE/FRND/FN/BI-01-C       PASS
MESH    MESH/NODE/FRND/FN/BI-02-C       PASS
MESH    MESH/NODE/FRND/FN/BI-03-C       PASS
MESH    MESH/NODE/FRND/LPN/BV-01-C      PASS
MESH    MESH/NODE/FRND/LPN/BV-02-C      INCONC
MESH    MESH/NODE/FRND/LPN/BV-03-C      PASS
MESH    MESH/NODE/FRND/LPN/BV-04-C      PASS
MESH    MESH/NODE/FRND/LPN/BV-05-C      PASS
MESH    MESH/NODE/FRND/LPN/BV-06-C      PASS
MESH    MESH/NODE/FRND/LPN/BV-07-C      PASS
MESH    MESH/NODE/FRND/LPN/BV-08-C      PASS
MESH    MESH/NODE/FRND/LPN/BI-01-C      PASS
MESH    MESH/NODE/FRND/LPN/BI-02-C      PASS
MESH    MESH/NODE/FRND/LPN/BI-03-C      PASS
MESH    MESH/SR/PROX/BV-01-C            PASS
MESH    MESH/SR/PROX/BV-02-C            PASS
MESH    MESH/SR/PROX/BV-03-C            PASS
MESH    MESH/SR/PROX/BV-04-C            PASS (2)
MESH    MESH/SR/PROX/BV-05-C            PASS
MESH    MESH/SR/PROX/BV-06-C            PASS
MESH    MESH/SR/PROX/BV-07-C            PASS
MESH    MESH/SR/PROX/BV-08-C            PASS
MESH    MESH/SR/PROX/BV-09-C            PASS
MESH    MESH/SR/PROX/BV-10-C            PASS
MESH    MESH/SR/PROX/BV-11-C            PASS
MESH    MESH/SR/PROX/BV-12-C            PASS
MESH    MESH/SR/PROX/BV-13-C            PASS
MESH    MESH/SR/PROX/BV-14-C            INCONC
MESH    MESH/SR/PROX/BV-15-C            PASS
MESH    MESH/SR/PROX/BI-01-C            PASS
MESH    MESH/SR/PROX/BI-02-C            PASS
MESH    MESH/SR/MPXS/BV-01-C            PASS
MESH    MESH/SR/MPXS/BV-02-C            PASS
MESH    MESH/SR/MPXS/BV-03-C            PASS
MESH    MESH/SR/MPXS/BV-04-C            PASS
MESH    MESH/SR/MPXS/BV-05-C            PASS
MESH    MESH/SR/MPXS/BV-06-C            PASS
MESH    MESH/SR/MPXS/BV-07-C            PASS
MESH    MESH/SR/MPXS/BV-09-C            INCONC
MESH    MESH/NODE/CFG/SNBP/BV-01-C      PASS
MESH    MESH/NODE/CFG/SNBP/BI-01-C      PASS
MESH    MESH/NODE/CFG/COMP/BV-01-C      PASS
MESH    MESH/NODE/CFG/DTTL/BV-01-C      PASS
MESH    MESH/NODE/CFG/DTTL/BI-01-C      PASS
MESH    MESH/NODE/CFG/GPXY/BV-01-C      PASS
MESH    MESH/NODE/CFG/GPXY/BV-02-C      PASS
MESH    MESH/NODE/CFG/GPXY/BI-01-C      PASS
MESH    MESH/NODE/CFG/CFGF/BV-01-C      PASS
MESH    MESH/NODE/CFG/CFGF/BI-01-C      PASS
MESH    MESH/NODE/CFG/CFGR/BV-01-C      PASS
MESH    MESH/NODE/CFG/CFGR/BI-01-C      PASS
MESH    MESH/NODE/CFG/MP/BV-01-C        PASS
MESH    MESH/NODE/CFG/MP/BV-02-C        PASS
MESH    MESH/NODE/CFG/MP/BV-03-C        PASS
MESH    MESH/NODE/CFG/MP/BI-01-C        PASS
MESH    MESH/NODE/CFG/MP/BI-02-C        PASS
MESH    MESH/NODE/CFG/MP/BI-03-C        PASS
MESH    MESH/NODE/CFG/SL/BV-01-C        PASS
MESH    MESH/NODE/CFG/SL/BV-02-C        PASS
MESH    MESH/NODE/CFG/SL/BV-03-C        PASS
MESH    MESH/NODE/CFG/SL/BI-01-C        PASS
MESH    MESH/NODE/CFG/SL/BI-02-C        PASS
MESH    MESH/NODE/CFG/SL/BI-03-C        PASS
MESH    MESH/NODE/CFG/SL/BI-04-C        PASS
MESH    MESH/NODE/CFG/SL/BI-05-C        PASS
MESH    MESH/NODE/CFG/SL/BI-06-C        PASS
MESH    MESH/NODE/CFG/SL/BI-07-C        PASS
MESH    MESH/NODE/CFG/SL/BI-08-C        PASS
MESH    MESH/NODE/CFG/SL/BI-09-C        PASS
MESH    MESH/NODE/CFG/SL/BI-10-C        PASS
MESH    MESH/NODE/CFG/NKL/BV-01-C       PASS
MESH    MESH/NODE/CFG/NKL/BV-02-C       PASS
MESH    MESH/NODE/CFG/NKL/BV-04-C       PASS
MESH    MESH/NODE/CFG/NKL/BI-01-C       PASS
MESH    MESH/NODE/CFG/NKL/BI-02-C       PASS
MESH    MESH/NODE/CFG/NKL/BI-03-C       PASS
MESH    MESH/NODE/CFG/AKL/BV-01-C       PASS
MESH    MESH/NODE/CFG/AKL/BV-02-C       PASS
MESH    MESH/NODE/CFG/AKL/BV-03-C       PASS
MESH    MESH/NODE/CFG/AKL/BI-01-C       PASS
MESH    MESH/NODE/CFG/AKL/BI-02-C       PASS
MESH    MESH/NODE/CFG/AKL/BI-03-C       PASS
MESH    MESH/NODE/CFG/AKL/BI-04-C       PASS
MESH    MESH/NODE/CFG/MAKL/BV-01-C      PASS
MESH    MESH/NODE/CFG/MAKL/BV-02-C      PASS
MESH    MESH/NODE/CFG/MAKL/BI-01-C      PASS
MESH    MESH/NODE/CFG/MAKL/BI-02-C      PASS
MESH    MESH/NODE/CFG/MAKL/BI-03-C      PASS
MESH    MESH/NODE/CFG/MAKL/BI-04-C      PASS
MESH    MESH/NODE/CFG/NID/BV-01-C       PASS
MESH    MESH/NODE/CFG/NID/BV-02-C       PASS
MESH    MESH/NODE/CFG/NID/BV-03-C       PASS
MESH    MESH/NODE/CFG/NID/BI-01-C       PASS
MESH    MESH/NODE/CFG/NID/BI-03-C       PASS
MESH    MESH/NODE/CFG/RST/BV-01-C       PASS
MESH    MESH/NODE/CFG/HBP/BV-01-C       PASS
MESH    MESH/NODE/CFG/HBP/BV-02-C       PASS
MESH    MESH/NODE/CFG/HBP/BV-03-C       PASS
MESH    MESH/NODE/CFG/HBP/BV-04-C       PASS
MESH    MESH/NODE/CFG/HBP/BV-05-C       PASS
MESH    MESH/NODE/CFG/HBP/BV-06-C       PASS
MESH    MESH/NODE/CFG/HBP/BV-07-C       PASS
MESH    MESH/NODE/CFG/HBP/BI-01-C       PASS
MESH    MESH/NODE/CFG/HBP/BI-02-C       PASS
MESH    MESH/NODE/CFG/HBS/BV-01-C       PASS
MESH    MESH/NODE/CFG/HBS/BV-02-C       PASS
MESH    MESH/NODE/CFG/HBS/BV-03-C       PASS
MESH    MESH/NODE/CFG/HBS/BV-04-C       PASS
MESH    MESH/NODE/CFG/HBS/BV-05-C       PASS (3)
MESH    MESH/NODE/CFG/HBS/BI-01-C       PASS
MESH    MESH/NODE/CFG/NTX/BV-01-C       PASS
MESH    MESH/NODE/CFG/LPNPT/BI-01-C     PASS
MESH    MESH/SR/HM/CFS/BV-01-C          PASS
MESH    MESH/SR/HM/CFS/BV-02-C          PASS
MESH    MESH/SR/HM/RFS/BV-01-C          PASS
MESH    MESH/SR/HM/RFS/BV-02-C          PASS
MESH    MESH/SR/HM/RFS/BV-03-C          PASS
MESH    MESH/SR/HM/RFS/BI-01-C          PASS
MESH    MESH/SR/HM/RFS/BI-02-C          PASS
MESH    MESH/SR/HM/HPS/BV-01-C          PASS
MESH    MESH/SR/HM/ATS/BV-01-C          PASS
</pre>
</details>

**Comment by KKopyscinski** [comment]
Rebased onto master to resolve conflict

**Comment by michal-narajowski** [comment]
@KKopyscinski Please see CI results. `nordic_pca10056_blemesh_cdb` is not compiling as well as porting targets.

File: nimble/host/mesh/include/mesh/health_cli.h [hunk 1]
```diff
@@ -29,9 +29,7 @@ struct bt_mesh_health_cli {
 			       uint8_t test_id, uint16_t cid, uint8_t *faults,
 			       size_t fault_count);
 
-	struct k_sem          op_sync;
-	uint32_t                 op_pending;
-	void                 *op_param;
+	struct bt_mesh_msg_ack_ctx ack_ctx;
```
Comment (michal-narajowski) [inline nimble/host/mesh/include/mesh/health_cli.h:32]: Indentation.

File: nimble/host/mesh/src/access.c [hunk 2]
```diff
@@ -171,67 +194,68 @@ static int publish_retransmit(struct bt_mesh_model *mod)
...
 
 static void mod_publish(struct ble_npl_event *work)
 {
 	struct bt_mesh_model_pub *pub = ble_npl_event_get_arg(work);
-	int32_t period_ms;
 	int err;
 
-	BT_DBG("");
+	if (pub->addr == BT_MESH_ADDR_UNASSIGNED ||
+	atomic_test_bit(bt_mesh.flags, BT_MESH_SUSPENDED)) {
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/access.c:232]: Indetntation?

File: nimble/host/mesh/src/access.c [hunk 3]
```diff
@@ -431,32 +484,75 @@ struct bt_mesh_elem *bt_mesh_elem_find(uint16_t addr)
...
 
-	return NULL;
+	return false;
+}
+
+#if MYNEWT_VAL(BLE_MESH_ACCESS_LAYER_MSG)
+void bt_mesh_msg_cb_set(void (*cb)(uint32_t opcode, struct bt_mesh_msg_ctx *ctx,
+			struct os_mbuf *buf))
+{
+	msg_cb = cb;
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/access.c:526]: Indentation.

File: nimble/host/mesh/src/adv_legacy.c [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+}
+
+int bt_mesh_adv_start(const struct ble_gap_adv_params *param, int32_t duration,
+		      const struct bt_data *ad, size_t ad_len,
+		      const struct bt_data *sd, size_t sd_len)
+{
+	adv_timeout = duration;
+	return bt_le_adv_start(param, duration, ad, ad_len, sd, sd_len);
+}
+#endif
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/adv_legacy.c:245]: Newline.

File: nimble/host/mesh/src/app_keys.c [hunk 5]
```diff
@@ -498,10 +629,89 @@ uint16_t bt_mesh_app_key_find(bool dev_key, uint8_t aid,
...
+{
+#if MYNEWT_VAL(BLE_MESH_SETTINGS)
+	int rc;
+
+	rc = conf_register(&bt_mesh_app_key_conf_handler);
+
+	SYSINIT_PANIC_ASSERT_MSG(rc == 0,
+				 "Failed to register bt_mesh_app_key conf");
+#endif
+}
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/app_keys.c:717]: Newline.

File: nimble/host/mesh/src/msg.c [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+	if (ack->op != op || (BT_MESH_ADDR_IS_UNICAST(ack->dst) && ack->dst != addr)) {
+		return false;
+	}
+
+	if (user_data != NULL) {
+		*user_data = ack->user_data;
+	}
+
+	return true;
+}
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/msg.c:84]: Newline.

File: nimble/host/mesh/src/pb_gatt.c [hunk 7]
```diff
@@ -155,4 +171,5 @@ const struct prov_bearer pb_gatt = {
 	.link_accept = link_accept,
 	.send = buf_send,
 	.clear_tx = clear_tx,
-};
\ No newline at end of file
+};
+#endif
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/pb_gatt.c:175]: Newline.

File: nimble/host/mesh/src/pb_gatt_srv.c [hunk 8]
```diff
@@ -0,0 +1,471 @@
...
+	err = bt_mesh_adv_start(&fast_adv_param, (60 * MSEC_PER_SEC),
+				prov_ad, ARRAY_SIZE(prov_ad),
+				prov_sd, prov_sd_len);
+	if (!err) {
+		prov_fast_adv = false;
+	}
+
+	return err;
+}
+#endif
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/pb_gatt_srv.c:471]: Newline.

File: nimble/host/mesh/src/pb_gatt_srv.h [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+extern struct svc_handles {
+	uint16_t proxy_h;
+	uint16_t proxy_data_out_h;
+	uint16_t prov_h;
+	uint16_t prov_data_in_h;
+	uint16_t prov_data_out_h;
+} svc_handles;
+
+#endif /* __PB_GATT_SRV_H__ */
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/pb_gatt_srv.h:31]: Newline.

File: nimble/host/mesh/syscfg.yml [hunk 10]
```diff
@@ -879,3 +975,8 @@ syscfg.vals.BLE_MESH_PB_GATT:
 
 syscfg.vals.BLE_MESH_PB_ADV:
     BLE_MESH_PROV: 1
+
+syscfg.vals.'BLE_MESH_PB_GATT':
+    BLE_MESH_PROXY_MSG_LEN: 66
+syscfg.vals.'BLE_MESH_GATT_PROXY':
+    BLE_MESH_PROXY_MSG_LEN: 33
```
Comment (michal-narajowski) [inline nimble/host/mesh/syscfg.yml:982]: Newline.

File: nimble/host/mesh/src/transport.h [hunk 11]
```diff
@@ -102,12 +95,12 @@ void bt_mesh_trans_init(void);
...
 
 uint8_t bt_mesh_va_del(const uint8_t uuid[16], uint16_t *addr);
 
-uint8_t *bt_mesh_va_label_get(uint16_t addr);
\ No newline at end of file
+uint8_t *bt_mesh_va_label_get(uint16_t addr);
+
+void bt_mesh_va_pending_store(void);
+
+void bt_mesh_va_init(void);
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/transport.h:106]: Newline.

**Review by michal-narajowski — APPROVED** [review]

**Pull request merged**